packages/fetchai/skills/gym/handlers.py

```diff
@@ -41,6 +42,12 @@ def __init__(self, **kwargs):
         logger.info("GymHandler.__init__: arguments: {}".format(kwargs))
         super().__init__(**kwargs)
 
+        nb_steps = kwargs.get("nb_steps", DEFAULT_NB_STEPS)
+        self.task = GymTask(self.context, nb_steps)
+        # launch the task
+        self.task.setup()
+        self.task.execute()
```

Comment:
instead of submitting the task to the task manager, we instantiate the `GymTask`, which in turn create a thread (as before) to run the fit of the agent.

We can't submit this to the multiprocess version of the TaskManager, since the ProxyEnv.step() (called by the RLAgent.fit) cannot put messages to the outbox.